library/include/borealis/core/view.hpp

```diff
@@ -1032,6 +1034,35 @@ class View
         return this->actions;
     }
 
+    /**
+     * Get the vector of all gesture recognizers attached to that view.
+     */
+    const std::vector<GestureRecognizer*>& getGestureRecognizers()
+    {
+        return this->gestureRecognizers;
+    }
+
+    /**
+     * Interrupt every recognizer on this view.
+     * If onlyIfUnsureState == true, only recognizers with
+     * current state UNSURE will be interupted
+     */
+    void interruptGestures(bool onlyIfUnsureState);
+
+    /**
+     * Add new gesture recognizer on this view.
+     */
+    void addGestureRecognizer(GestureRecognizer* recognizer);
+
+    /**
+     * Called each frame when touch is registered.
+     * 
+     * @return false if there is no recognizer 
+     * that needs to play it's own click sound, 
+     * so play default is allowed
+     */
+    bool gestureRecognizerRequest(Touch touch, View* firstResponder);
```

Comment:
I feel the same about the "return false to play a sound", it should be passed as a parameter, maybe a "sound action" enum like PLAY_DEFAULT, DONT_PLAY, etc...